azimuth/modules/dataset_analysis/dataset_warnings.py

```diff
@@ -71,54 +72,32 @@ def get_general_warnings(
         Returns:
             General warnings
         """
+        first_dm = next(iter(dm_dict.values()))
+        cls_names = first_dm.get_class_names(labels_only=True)
+
         count_per_cls_per_split = {
             s: dm.class_distribution(labels_only=True) for s, dm in dm_dict.items()
         }
 
-        # Compute all data based on class distribution needed for all general warnings.
+        # Missing samples warning
         min_count_per_cls = self.config.dataset_warnings.min_num_per_class
-        max_delta_imb = self.config.dataset_warnings.max_delta_class_imbalance
-        alert_min_per_cls_per_split = {}
-        mean_per_split = {}
-        imb_per_cls_per_split = {}
-        count_norm_per_cls_per_split = {}
+        alert_min_per_cls_per_split: Dict[DatasetSplitName, np.ndarray] = defaultdict(
+            partial(np.full, shape=len(cls_names), fill_value=False)
+        )
         for split, count in count_per_cls_per_split.items():
             alert_min_per_cls_per_split[split] = count < min_count_per_cls
-            mean_per_split[split] = np.mean(count)
-            imb_per_cls_per_split[split] = count / mean_per_split[split] - 1
-            count_norm_per_cls_per_split[split] = count / sum(count)
 
-        # Missing samples warning
         alert_min_per_cls = (
             alert_min_per_cls_per_split[DatasetSplitName.train]
             | alert_min_per_cls_per_split[DatasetSplitName.eval]
         )
```

Comment:
SImilarly, this
```suggestion
        alert_min_per_cls = np.logical_or.reduce(tuple(alert_min_per_cls_per_split.values()))
```
would allow for a normal `dict` like <following comment>